Orange/tests/test_statistics.py

```diff
@@ -241,3 +228,94 @@ def test_var(self):
+        x = csr_matrix([0, 1, 0, 2, 2, np.nan, 1, np.nan, 0, 1])
+        self.assertEqual(countnans(x), 2)
+
+    def test_shape_matches_dense_and_sparse_given_array_and_axis_None(self):
+        dense = np.array([0, 1, 0, 2, 2, np.nan, 1, np.nan, 0, 1])
+        sparse = csr_matrix(dense)
+        np.testing.assert_equal(countnans(dense), countnans(sparse))
+        self.assertEqual(countnans(dense), 2)
+
+    def test_shape_matches_dense_and_sparse_given_array_and_axis_0(self):
+        dense = np.array([0, 1, 0, 2, 2, np.nan, 1, np.nan, 0, 1])
+        sparse = csr_matrix(dense)
+        np.testing.assert_equal(countnans(dense, axis=0), countnans(sparse, axis=0))
+        self.assertEqual(countnans(dense, axis=0), 2)
+
+    def test_shape_matches_dense_and_sparse_given_array_and_axis_1(self):
+        dense = np.array([0, 1, 0, 2, 2, np.nan, 1, np.nan, 0, 1])
+        sparse = csr_matrix(dense)
+        np.testing.assert_equal(countnans(dense, axis=1), countnans(sparse, axis=1))
+        self.assertEqual(countnans(dense, axis=1), 2)
+
+    def test_countnans_on_2d_matrix(self):
+        x = [[1, np.nan],
+             [2, np.nan]]
+        np.testing.assert_equal(
+            countnans(x), 2, 'Countnans fails on dense data')
+        np.testing.assert_equal(
+            countnans(csr_matrix(x)), 2, 'Countnans fails on sparse data.')
+
+    def test_countnans_on_columns(self):
+        x = [[1, np.nan],
+             [2, np.nan]]
+        np.testing.assert_equal(
+            countnans(x, axis=0), [0, 2],
+            'Countnans fails on dense data with `axis=0`')
+        np.testing.assert_equal(
+            countnans(csr_matrix(x), axis=0), [0, 2],
+            'Countnans fails on sparse data with `axis=0`')
+
+    def test_countnans_on_rows(self):
+        x = [[1, np.nan],
+             [2, np.nan]]
+        np.testing.assert_equal(
+            countnans(x, axis=1), [1, 1],
+            'Countnans fails on dense data with `axis=1`')
+        np.testing.assert_equal(
+            countnans(csr_matrix(x), axis=1), [1, 1],
+            'Countnans fails on sparse data with `axis=1`')
+
+    def test_countnans_with_2d_weights(self):
+        x = [[1, np.nan],
+             [2, np.nan]]
+        w = np.array([[1, 1],
+                      [2, 2]])
+        np.testing.assert_equal(countnans(x, weights=w, axis=0), [0, 3])
+        np.testing.assert_equal(countnans(x, weights=w, axis=1), [1, 2])
+
+        w = np.array([1, 2])
+        np.testing.assert_equal(countnans(x, weights=w, axis=0), [0, 4])
+        np.testing.assert_equal(countnans(x, weights=w, axis=1), [1, 2])
+
+
+class TestBincount(unittest.TestCase):
+    def test_bincount_counts_nans(self):
+        dense = [0, 0, 0, 1, np.nan, 1]
+        sparse = csr_matrix(dense)
+        expected = [3, 2], 1
+
+        np.testing.assert_equal(bincount(dense), expected)
+        np.testing.assert_equal(bincount(sparse), expected)
+
+        hist, n_nans = bincount([0., 1., 3], max_val=3)
```

Comment:
What exactly does `max_val=3` does here?

Reply:
This was a remnant of the old tests. I've removed it and added an appropriate test.